source/adaptive_load/step_controller_impl.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include "nighthawk/adaptive_load/input_variable_setter.h"
+#include "nighthawk/adaptive_load/step_controller.h"
+
+#include "api/adaptive_load/adaptive_load.pb.h"
+#include "api/adaptive_load/benchmark_result.pb.h"
+#include "api/adaptive_load/step_controller_impl.pb.h"
+#include "envoy/registry/registry.h"
+
+namespace Nighthawk {
+
+/**
+ * A StepController that performs an exponential search for the highest load that keeps metrics
+ * within thresholds.
+ */
+class ExponentialSearchStepController : public StepController {
+public:
+  explicit ExponentialSearchStepController(
+      const nighthawk::adaptive_load::ExponentialSearchStepControllerConfig& config,
+      nighthawk::client::CommandLineOptions command_line_options_template);
+  absl::StatusOr<nighthawk::client::CommandLineOptions>
+  GetCurrentCommandLineOptions() const override;
+  bool IsConverged() const override;
+  bool IsDoomed(std::string& doom_reason) const override;
+  void UpdateAndRecompute(const nighthawk::adaptive_load::BenchmarkResult& result) override;
+
+private:
+  const nighthawk::client::CommandLineOptions command_line_options_template_;
+  InputVariableSetterPtr input_variable_setter_;
+  bool is_exponential_phase_;
```

Comment:
This is actually the standard name, if it can be called that. I was surprised to find out about the binary search that is part of it: https://en.wikipedia.org/wiki/Exponential_search